commit d107fe25b44313b4ac480c82a5e26833aa73e408
Author: Rene Kjellerup <rk.katana.steel@gmail.com>
Date:   Tue Oct 22 13:17:04 2019 -0700

    upgrade the GUI client to QT5
    
    and updated the readme documentation to reflect
    the new software requirements

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8170453..a7b70c4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -527,17 +527,15 @@ if( QTGUI )
 
 	add_definitions( -DQT_THREAD_SUPPORT )
 
-	find_package(
-		Qt4
-		COMPONENTS QtCore QtGui REQUIRED )
+	find_package(Qt5 COMPONENTS Core Widgets REQUIRED )
 
-	if( NOT QT_QTGUI_FOUND )
+	if( NOT Qt5Widgets_FOUND )
 
 		message(
 			FATAL_ERROR
 			"Unable to locate required package : QT" )
 
-	endif( NOT QT_QTGUI_FOUND )
+	endif( NOT Qt5Widgets_FOUND )
 		
 	# success
 
diff --git a/README.TXT b/README.TXT
index 813af7c..ccb771e 100644
--- a/README.TXT
+++ b/README.TXT
@@ -66,7 +66,7 @@ lex or flex
 libedit
 bison >= 2.3
 cmake >= 2.4
-opsenssl >= 0.9.x
+opsenssl >= 1.0.x
 
 This software requires Bison version 2.3 or higher as it uses the new c++
 parser model. The c++ model was introduced in version 2.1 but appears to be
@@ -75,7 +75,7 @@ incomplete.
 Optional
 --------------------------------------------------------------------------------
 
-TrollTech QT >= 4.2
+TrollTech QT >= 5.0
 openldap >= 2.0
 
 The Trolltech QT toolkit version 4.2.x or higher is also required if you plan
diff --git a/source/iked/CMakeLists.txt b/source/iked/CMakeLists.txt
index 2e41586..bad2971 100644
--- a/source/iked/CMakeLists.txt
+++ b/source/iked/CMakeLists.txt
@@ -7,6 +7,8 @@
 #        : Copyright 2007, Shrew Soft Inc
 #
 
+cmake_policy(SET CMP0050 OLD)
+
 add_definitions( -D PATH_CONF=\\"${PATH_ETC}\\" )
 
 include_directories(
@@ -109,18 +111,18 @@ add_custom_target( ConfParser echo "Creating Conf Parser" )
 # Custom command for flex
 
 add_custom_command(
-	SOURCE conf.token.ll
+	SOURCE ${CMAKE_CURRENT_SOURCE_DIR}/conf.token.ll
 	COMMAND ${PATH_BIN_FLEX}
-	ARGS -d -oconf.token.cpp conf.token.ll
+	ARGS -d -oconf.token.cpp ${CMAKE_CURRENT_SOURCE_DIR}/conf.token.ll
 	TARGET ConfParser
 	OUTPUTS conf.token.cpp )
 
 # Custom command for bison
 
 add_custom_command(
-	SOURCE conf.parse.yy
+	SOURCE ${CMAKE_CURRENT_SOURCE_DIR}/conf.parse.yy
 	COMMAND ${PATH_BIN_BISON}
-	ARGS -oconf.parse.cpp conf.parse.yy
+	ARGS -oconf.parse.cpp ${CMAKE_CURRENT_SOURCE_DIR}/conf.parse.yy
 	TARGET ConfParser
 	OUTPUTS conf.parse.cpp conf.parse.hpp location.hh position.hh stack.hh )
 
diff --git a/source/qikea/CMakeLists.txt b/source/qikea/CMakeLists.txt
index 98682e2..5d95f97 100644
--- a/source/qikea/CMakeLists.txt
+++ b/source/qikea/CMakeLists.txt
@@ -14,15 +14,11 @@ include_directories(
 	${IKE_SOURCE_DIR}/source/libidb
 	${IKE_SOURCE_DIR}/source/libith
 	${IKE_SOURCE_DIR}/source/liblog
-	${QT_INCLUDES} )
+        ${CMAKE_CURRENT_BINARY_DIR} )
 
-link_directories(
-	${IKE_SOURCE_DIR}/source/libip
-	${QT_LIBRARY_DIR} )
+link_directories( ${IKE_SOURCE_DIR}/source/libip )
 
-include( ${QT_USE_FILE} )
-
-qt4_wrap_ui(
+qt5_wrap_ui(
 	UIH_FILES
 	about.ui
 	conflict.ui
@@ -30,11 +26,11 @@ qt4_wrap_ui(
 	site.ui
 	topology.ui )
 
-qt4_wrap_cpp(
+qt5_wrap_cpp(
 	CXX_FILES
 	qikea.h )
 
-qt4_add_resources(
+qt5_add_resources(
 	QRC_FILES
 	qikea.qrc )
 
@@ -59,6 +55,8 @@ target_link_libraries(
 	pthread
 	${QT_LIBRARIES} )
 
+qt5_use_modules ( qikea Widgets )
+
 if( APPLE )
 
 	set(
diff --git a/source/qikea/root.cpp b/source/qikea/root.cpp
index 97a8345..4328102 100644
--- a/source/qikea/root.cpp
+++ b/source/qikea/root.cpp
@@ -45,7 +45,7 @@ bool file_to_bdata( QString path, BDATA & bdata )
 {
 	bdata.del();
 
-	FILE * fp = fopen( path.toAscii(), "rb" );
+	FILE * fp = fopen( path.toUtf8().constData(), "rb" );
 	if( fp == NULL )
 		return false;
 
@@ -65,7 +65,7 @@ bool file_to_bdata( QString path, BDATA & bdata )
 
 bool bdata_to_file( BDATA & bdata, QString path )
 {
-	FILE * fp = fopen( path.toAscii(), "wb" );
+	FILE * fp = fopen( path.toUtf8().constData(), "wb" );
 	if( fp == NULL )
 		return false;
 
@@ -87,7 +87,7 @@ void site_name_mangle( QString & name )
 	QString tmpName = name;
 
 	CONFIG tmpConfig;
-	tmpConfig.set_id( name.toAscii() );
+	tmpConfig.set_id( name.toUtf8().constData() );
 
 	int index = 2;
 
@@ -97,7 +97,7 @@ void site_name_mangle( QString & name )
 		name += QString( " (" );
 		name += QString::number( index++ );
 		name += QString( ")" );
-		tmpConfig.set_id( name.toAscii() );
+		tmpConfig.set_id( name.toUtf8().constData() );
 	}
 }
 
@@ -220,7 +220,7 @@ void _qikeaRoot::siteAdd()
 	// mangle name if duplicate
 
 	site_name_mangle( siteName );
-	config.set_id( siteName.toAscii() );
+	config.set_id( siteName.toUtf8().constData() );
 
 	// write site config
 
@@ -249,7 +249,7 @@ void _qikeaRoot::siteModify()
 	// load site config
 
 	CONFIG config;
-	config.set_id( i->text().toAscii() );
+	config.set_id( i->text().toUtf8().constData() );
 	qikea.manager.file_vpn_load( config );
 
 	// create site modal dialog
@@ -280,7 +280,7 @@ void _qikeaRoot::siteDelete()
 		return;
 
 	CONFIG config;
-	config.set_id( i->text().toAscii() );
+	config.set_id( i->text().toUtf8().constData() );
 	qikea.manager.file_vpn_del( config );
 
 	delete i;
@@ -312,11 +312,11 @@ void _qikeaRoot::siteRenamed( QListWidgetItem * i )
 	site_name_mangle( modName );
 
 	CONFIG config;
-	config.set_id( oldName.toAscii() );
+	config.set_id( oldName.toUtf8().constData() );
 	qikea.manager.file_vpn_load( config );
 	qikea.manager.file_vpn_del( config );
 
-	config.set_id( modName.toAscii() );
+	config.set_id( modName.toUtf8().constData() );
 	qikea.manager.file_vpn_save( config );
 
 	i->setText( modName );
@@ -346,9 +346,9 @@ void _qikeaRoot::siteImport()
 	bool need_certs = false;
 
 	if( !loadPath.contains( ".pcf", Qt::CaseInsensitive ) )
-		qikea.manager.file_vpn_load( config, loadPath.toAscii(), false );
+		qikea.manager.file_vpn_load( config, loadPath.toUtf8().constData(), false );
 	else
-		qikea.manager.file_pcf_load( config, loadPath.toAscii(), need_certs );
+		qikea.manager.file_pcf_load( config, loadPath.toUtf8().constData(), need_certs );
 
 	// determine file name
 
@@ -361,7 +361,7 @@ void _qikeaRoot::siteImport()
 
 	// save the site config
 
-	config.set_id( siteName.toAscii() );
+	config.set_id( siteName.toUtf8().constData() );
 	qikea.manager.file_vpn_save( config );
 
 	// create icon for site
@@ -402,7 +402,7 @@ void _qikeaRoot::siteExport()
 	// load site config
 
 	CONFIG config;
-	config.set_id( i->text().toAscii() );
+	config.set_id( i->text().toUtf8().constData() );
 	qikea.manager.file_vpn_load( config );
 
 	// get the output path
@@ -419,7 +419,7 @@ void _qikeaRoot::siteExport()
 	if( !savePath.length() )
 		return;
 
-	qikea.manager.file_vpn_save( config, savePath.toAscii() );
+	qikea.manager.file_vpn_save( config, savePath.toUtf8().constData() );
 }
 
 void _qikeaRoot::showAbout()
diff --git a/source/qikea/site.cpp b/source/qikea/site.cpp
index 4b7f57c..e32a852 100644
--- a/source/qikea/site.cpp
+++ b/source/qikea/site.cpp
@@ -837,7 +837,7 @@ bool _qikeaSite::save( CONFIG & config )
 	// remote name or address
 
 	config.set_string( "network-host",
-		lineEditHost->text().toAscii(),
+		lineEditHost->text().toUtf8().constData(),
 		lineEditHost->text().length() );
 
 	// remote ike port
@@ -929,7 +929,7 @@ bool _qikeaSite::save( CONFIG & config )
 			Address = Address.replace( ' ', "" );
 
 			config.set_string( "client-ip-addr",
-				Address.toAscii(),
+				Address.toUtf8().constData(),
 				Address.length() );
 
 			// adapter netmask
@@ -938,7 +938,7 @@ bool _qikeaSite::save( CONFIG & config )
 			Netmask = Netmask.replace( ' ', "" );
 
 			config.set_string( "client-ip-mask",
-				Netmask.toAscii(),
+				Netmask.toUtf8().constData(),
 				Netmask.length() );
 		}
 	}
@@ -948,7 +948,7 @@ bool _qikeaSite::save( CONFIG & config )
 	// nat traversal mode
 
 	config.set_string( "network-natt-mode",
-		comboBoxNATTMode->currentText().toAscii(),
+		comboBoxNATTMode->currentText().toUtf8().constData(),
 		comboBoxNATTMode->currentText().length() );
 
 	// natt not disabled
@@ -971,7 +971,7 @@ bool _qikeaSite::save( CONFIG & config )
 	// ike fragment mode
 
 	config.set_string( "network-frag-mode",
-		comboBoxFragMode->currentText().toAscii(),
+		comboBoxFragMode->currentText().toUtf8().constData(),
 		comboBoxFragMode->currentText().length() );
 
 	// ike frag not disabled
@@ -1040,33 +1040,33 @@ bool _qikeaSite::save( CONFIG & config )
 			DNSServer = lineEditDNSServer1->text();
 			DNSServer = DNSServer.replace( ' ', "" );
 
-			if( inet_addr( DNSServer.toAscii() ) )
+			if( inet_addr( DNSServer.toUtf8().constData() ) )
 				config.add_string( "client-dns-addr",
-					DNSServer.toAscii(),
+					DNSServer.toUtf8().constData(),
 					DNSServer.length() );
 
 			DNSServer = lineEditDNSServer2->text();
 			DNSServer = DNSServer.replace( ' ', "" );
 
-			if( inet_addr( DNSServer.toAscii() ) )
+			if( inet_addr( DNSServer.toUtf8().constData() ) )
 				config.add_string( "client-dns-addr",
-					DNSServer.toAscii(),
+					DNSServer.toUtf8().constData(),
 					DNSServer.length() );
 
 			DNSServer = lineEditDNSServer3->text();
 			DNSServer = DNSServer.replace( ' ', "" );
 
-			if( inet_addr( DNSServer.toAscii() ) )
+			if( inet_addr( DNSServer.toUtf8().constData() ) )
 				config.add_string( "client-dns-addr",
-					DNSServer.toAscii(),
+					DNSServer.toUtf8().constData(),
 					DNSServer.length() );
 
 			DNSServer = lineEditDNSServer4->text();
 			DNSServer = DNSServer.replace( ' ', "" );
 
-			if( inet_addr( DNSServer.toAscii() ) )
+			if( inet_addr( DNSServer.toUtf8().constData() ) )
 				config.add_string( "client-dns-addr",
-					DNSServer.toAscii(),
+					DNSServer.toUtf8().constData(),
 					DNSServer.length() );
 		}
 
@@ -1085,7 +1085,7 @@ bool _qikeaSite::save( CONFIG & config )
 			config.del( "client-dns-suffix" );
 
 			config.set_string( "client-dns-suffix",
-				lineEditDNSSuffix->text().toAscii(),
+				lineEditDNSSuffix->text().toUtf8().constData(),
 				lineEditDNSSuffix->text().length() );
 		}
 	}
@@ -1157,7 +1157,7 @@ bool _qikeaSite::save( CONFIG & config )
 
 	if( lineEditLocalIDData->isEnabled() )
 		config.set_string( "ident-client-data",
-			lineEditLocalIDData->text().toAscii(),
+			lineEditLocalIDData->text().toUtf8().constData(),
 			lineEditLocalIDData->text().length() );
 	else
 		config.del( "ident-client-data" );
@@ -1194,7 +1194,7 @@ bool _qikeaSite::save( CONFIG & config )
 
 	if( lineEditRemoteIDData->isEnabled() )
 		config.set_string( "ident-server-data",
-			lineEditRemoteIDData->text().toAscii(),
+			lineEditRemoteIDData->text().toUtf8().constData(),
 			lineEditRemoteIDData->text().length() );
 	else
 		config.del( "ident-server-data" );
@@ -1204,11 +1204,11 @@ bool _qikeaSite::save( CONFIG & config )
 	if( pathCAFile.size() )
 	{
 		config.set_string( "auth-server-cert-name",
-			lineEditCAName->text().toAscii(),
+			lineEditCAName->text().toUtf8().constData(),
 			lineEditCAName->text().length() );
 
 		BDATA fileData;
-		fileData.file_load( pathCAFile.toAscii() );
+		fileData.file_load( pathCAFile.toUtf8().constData() );
 		config.set_binary( "auth-server-cert-data",
 			fileData );
 	}
@@ -1224,11 +1224,11 @@ bool _qikeaSite::save( CONFIG & config )
 	if( pathCertFile.size() )
 	{
 		config.set_string( "auth-client-cert-name",
-			lineEditCertName->text().toAscii(),
+			lineEditCertName->text().toUtf8().constData(),
 			lineEditCertName->text().length() );
 
 		BDATA fileData;
-		fileData.file_load( pathCertFile.toAscii() );
+		fileData.file_load( pathCertFile.toUtf8().constData() );
 		config.set_binary( "auth-client-cert-data",
 			fileData );
 	}
@@ -1244,11 +1244,11 @@ bool _qikeaSite::save( CONFIG & config )
 	if( pathPKeyFile.size() )
 	{
 		config.set_string( "auth-client-key-name",
-			lineEditPKeyName->text().toAscii(),
+			lineEditPKeyName->text().toUtf8().constData(),
 			lineEditPKeyName->text().length() );
 
 		BDATA fileData;
-		fileData.file_load( pathPKeyFile.toAscii() );
+		fileData.file_load( pathPKeyFile.toUtf8().constData() );
 		config.set_binary( "auth-client-key-data",
 			fileData );
 	}
@@ -1265,7 +1265,7 @@ bool _qikeaSite::save( CONFIG & config )
 	{
 		BDATA psk;
 		psk.set(
-			( const char * ) lineEditPSK->text().toAscii(),
+			( const char * ) lineEditPSK->text().toUtf8().constData(),
 			lineEditPSK->text().length() );
 		config.set_binary( "auth-mutual-psk", psk );
 	}
@@ -1293,7 +1293,7 @@ bool _qikeaSite::save( CONFIG & config )
 	// phase1 cipher algorithm
 
 	config.set_string( "phase1-cipher",
-		comboBoxP1Cipher->currentText().toAscii(),
+		comboBoxP1Cipher->currentText().toUtf8().constData(),
 		comboBoxP1Cipher->currentText().length() );
 
 	// phase1 cipher key length
@@ -1304,7 +1304,7 @@ bool _qikeaSite::save( CONFIG & config )
 	// phase1 hash algorithm
 
 	config.set_string( "phase1-hash",
-		comboBoxP1Hash->currentText().toAscii(),
+		comboBoxP1Hash->currentText().toUtf8().constData(),
 		comboBoxP1Hash->currentText().length() );
 
 	// phase1 key life time
@@ -1327,7 +1327,7 @@ bool _qikeaSite::save( CONFIG & config )
 	// phase2 trasform algorithm
 
 	config.set_string( "phase2-transform",
-		comboBoxP2Transform->currentText().toAscii(),
+		comboBoxP2Transform->currentText().toUtf8().constData(),
 		comboBoxP2Transform->currentText().length() );
 
 	// phase2 transform key length
@@ -1338,7 +1338,7 @@ bool _qikeaSite::save( CONFIG & config )
 	// phase2 hmac algorithm
 
 	config.set_string( "phase2-hmac",
-		comboBoxP2HMAC->currentText().toAscii(),
+		comboBoxP2HMAC->currentText().toUtf8().constData(),
 		comboBoxP2HMAC->currentText().length() );
 
 	// phase2 pfs group
@@ -1360,13 +1360,13 @@ bool _qikeaSite::save( CONFIG & config )
 	// ipcomp transform algorithm
 
 	config.set_string( "ipcomp-transform",
-		comboBoxP2Compress->currentText().toAscii(),
+		comboBoxP2Compress->currentText().toUtf8().constData(),
 		comboBoxP2Compress->currentText().length() );
 
 	// policy level option
 
 	config.set_string( "policy-level",
-		comboBoxPolicyLevel->currentText().toAscii(),
+		comboBoxPolicyLevel->currentText().toUtf8().constData(),
 		comboBoxPolicyLevel->currentText().length() );
 
 	// policy nailed sa option
@@ -1409,7 +1409,7 @@ bool _qikeaSite::save( CONFIG & config )
 				// include
 
 				config.add_string( "policy-list-include",
-					i->text( 0 ).toAscii(),
+					i->text( 0 ).toUtf8().constData(),
 					i->text( 0 ).length() );
 			}
 
@@ -1418,7 +1418,7 @@ bool _qikeaSite::save( CONFIG & config )
 				// exlcude
 
 				config.add_string( "policy-list-exclude",
-					i->text( 0 ).toAscii(),
+					i->text( 0 ).toUtf8().constData(),
 					i->text( 0 ).length() );
 			}
 		}
@@ -1456,7 +1456,7 @@ bool _qikeaSite::verify()
 
 			QString Address = lineEditAddress->text();
 			Address = Address.replace( ' ', "" );
-			uint32_t addr = inet_addr( Address.toAscii() );
+			uint32_t addr = inet_addr( Address.toUtf8().constData() );
 
 			if( !addr || ( addr == INADDR_NONE ) )
 				errmsg = "Please enter valid virtual adapter address.";
@@ -1465,7 +1465,7 @@ bool _qikeaSite::verify()
 
 			QString Netmask = lineEditNetmask->text();
 			Netmask = Netmask.replace( ' ', "" );
-			uint32_t mask = inet_addr( Netmask.toAscii() );
+			uint32_t mask = inet_addr( Netmask.toUtf8().constData() );
 
 			if( !mask )
 				errmsg = "Please enter valid virtual adapter netmask.";
@@ -1487,7 +1487,7 @@ bool _qikeaSite::verify()
 
 			DNSServer = lineEditDNSServer1->text();
 			DNSServer = DNSServer.replace( ' ', "" );
-			addr = inet_addr( DNSServer.toAscii() );
+			addr = inet_addr( DNSServer.toUtf8().constData() );
 
 			if( addr && ( addr == INADDR_NONE ) )
 				errmsg = "Please enter valid DNS server #1 address.";
diff --git a/source/qikea/topology.cpp b/source/qikea/topology.cpp
index ba5a8d7..5efd10f 100644
--- a/source/qikea/topology.cpp
+++ b/source/qikea/topology.cpp
@@ -49,8 +49,8 @@ bool _qikeaTopology::verify()
 	QString Netmask = lineEditNetmask->text();
 	Netmask = Netmask.replace( ' ', "" );
 
-	in_addr_t addr = ntohl( inet_addr( Address.toAscii() ) );
-	in_addr_t mask = ntohl( inet_addr( Netmask.toAscii() ) );
+	in_addr_t addr = ntohl( inet_addr( Address.toUtf8().constData() ) );
+	in_addr_t mask = ntohl( inet_addr( Netmask.toUtf8().constData() ) );
 
 	// verify the netmask
 
diff --git a/source/qikec/CMakeLists.txt b/source/qikec/CMakeLists.txt
index 4a6f1b4..662032a 100644
--- a/source/qikec/CMakeLists.txt
+++ b/source/qikec/CMakeLists.txt
@@ -16,25 +16,22 @@ include_directories(
 	${IKE_SOURCE_DIR}/source/libith
 	${IKE_SOURCE_DIR}/source/liblog
 	${IKE_SOURCE_DIR}/source/libip
-	${QT_INCLUDES} )
+        ${CMAKE_CURRENT_BINARY_DIR} )
 
 link_directories(
-	${IKE_SOURCE_DIR}/source/libip
-	${QT_LIBRARY_DIR} )
-
-include( ${QT_USE_FILE} )
+	${IKE_SOURCE_DIR}/source/libip )
 
-qt4_wrap_ui(
+qt5_wrap_ui(
 	UIH_FILES
 	banner.ui
 	filepass.ui
 	root.ui )
 
-qt4_wrap_cpp(
+qt5_wrap_cpp(
 	CXX_FILES
 	qikec.h )
 
-qt4_add_resources(
+qt5_add_resources(
 	QRC_FILES
 	qikec.qrc )
 
@@ -57,6 +54,8 @@ target_link_libraries(
 	pthread
 	${QT_LIBRARIES} )
 
+qt5_use_modules( qikec Widgets )
+
 if( APPLE )
 
 	set(
diff --git a/source/qikec/qikec.cpp b/source/qikec/qikec.cpp
index d47b580..3f33ac7 100644
--- a/source/qikec/qikec.cpp
+++ b/source/qikec/qikec.cpp
@@ -105,7 +105,7 @@ bool _QIKEC::get_username()
 
 	username.del();
 	username.set(
-		( const char * ) data.text.toAscii(), data.text.length() );
+		( const char * ) data.text.toUtf8().constData(), data.text.length() );
 
 	return true;
 }
@@ -122,7 +122,7 @@ bool _QIKEC::get_password()
 
 	password.del();
 	password.set(
-		( const char * ) data.text.toAscii(), data.text.length() );
+		( const char * ) data.text.toUtf8().constData(), data.text.length() );
 
 	return true;
 }
@@ -144,7 +144,7 @@ bool _QIKEC::get_filepass( BDATA & path )
 	QString text = PassData.password;
 	fpass.del();
 	fpass.set(
-		( const char * ) text.toAscii(), text.length() );
+		( const char * ) text.toUtf8().constData(), text.length() );
 
 	return true;
 }
